### ISSUE TITLE
Use beginTime over endTime

### ISSUE THREAD

**Issue by ek-ex** [ta4j/ta4j#1213]
Bar Creation with End Time: I find the logic of creating bars with an EndTime and assigning the startTime in the bar constructor as endTime - duration a bit confusing. Typically, bar creation is done by specifying the bar's start time, which is also the standard across various platforms, market data providers, brokers, and exchanges. This approach may even constitute a bug, particularly for intraday time frames. Converting my data feed's start time to end time feels computationally inefficient and cumbersome. Currently, I add the duration to my datetime to correctly set the beginTime when creating a bar, which is an extra, unnecessary operation. How are others handling endTime?

The idea is to change the bars to favor begin time over end time. 

- Constructing the candles by passing the beginTime. 
- Print the candle using the beginTime. 
- Remove the endTime Instant from the bar to save memory and return beginTime + duration. 

let me know your comments and I can prepare a PR for this.

**Comment by ek-ex** [comment]
I'll incorporate the renaming of the BaseBar in a separate PR since this one is already big.

**Comment by TheCookieLab** [comment]
While identifying OHLC data with the start time is prevalent among crypto
exchanges, that's not necessarily the case within trad-fi where end of
period reporting has traditionally been the norm. If the tickers used in
the ta4j examples project are any indication, then trad-fi is the context
from where Ta4j's timestamp convention originated. Ta4j should have no
opinion and this is exactly the type of issue that the new project
vision/road-map would de-scope and push down to the client applications.

**Comment by ek-ex** [comment]
@TheCookieLab  I really liking the vision presented https://github.com/ta4j/ta4j-wiki/blob/master/Roadmap-and-Tasks.md
This would generalize the lib for each's usecase. Thank you and kudos.

Not just Crypto, stocks, options, futures all use beginTime.

Concretely when you say `push down to the client applications.` does that mean that the Bar Interface would support both values and up to the client which to use?

**Comment by if6was9** [comment]
I don't think the internal representation is a problem.  

It's the constructor/builder that is counter-intuitive and inconsistent with most data feeds. It does end up creating a bit of confusion for new users and off-by-one errors during development.  

Easily fixed with alternate constructor/builder, which I suspect most people using TA4J have done.

Just add beginTime to the constructor/builder and this problem is solved. Perhaps throw an exception if both beingTime and endTime are specified so that user doesn't have to think very hard.

**Comment by TheCookieLab** [comment]
Agreed, updating the constructor/builder with the alternative of setting the beginTime is a good solution.
